### PR TITLE
Fix conversion from ALTO to PAGE and vice versa

### PR DESCRIPTION
**Pull request by stweil**
- Fix order of arguments passed
- Remove shell debugging (-x)
- Handle input from STDIN
- Add `-convert-to ALTO` argument needed for conversion from PAGE to ALTO

### CODE REVIEW

**Comment by stweil** [comment]
The prima-page-converter does not format the output file. Should we add post processing which formats it nicely? Otherwise the web interface only shows a single (very lengthy) line.

**Comment by zuphilip** [comment]
Yes, pretty print would be good :+1: Can we use Saxon for that? Would it maybe even make sense to have a CLI option for that in ocr-transform?

I tried some examples in the Web GUI and see an error for alto__page transformation with https://rawgit.com/kba/ocr-fileformat-samples/master/samples/alto/2.0/wetzel_reisebegleiter_1901_0021.alto as well as with http://chroniclingamerica.loc.gov/lccn/sn86069133/1910-10-31/ed-1/seq-1/ocr.xml . But the third example works fine. Are these upstream problems?

**Comment by zuphilip** [comment]
The transformation GCV2hocr with the test file https://raw.githubusercontent.com/kba/ocr-fileformat-samples/master/samples/google-cloud/test_picture.json has an error message now, but maybe this is also unrelated to your PR here.

**Comment by zuphilip** [comment]
There seems to be a side effect in the abbyy2hocr transformation. In the Web GUI this transformation output nothing with the input https://digi.bib.uni-mannheim.de/~stweil/ocr-praxis/Testseiten/abbyy/417576986_0031.xml . However, this works in the docker run on the master branch. (Sorry for the the previous message which was unrelated to this issue.)

**Comment by stweil** [comment]
> There seems to be a side effect in the abbyy2hocr transformation.

That's strange because ABBYY to hOCR does not use a transformer script (and this PR does not change anything else).

**Comment by zuphilip** [comment]
Okay, I did another test with this branch here and can confirm that abbyy2hocr works fine in my docker container. There seems to be a problem with the instance on digi.

**Comment by stweil** [comment]
> Yes, pretty print would be good Can we use Saxon for that? Would it maybe even make sense to have a CLI option for that in ocr-transform?

There are a [lot of options](https://stackoverflow.com/questions/16090869/how-to-pretty-print-xml-from-the-command-line) how to implement pretty printing. I added a commit which uses Saxon, so the usual command line argument can be used to enable it (currently only implemented for output to STDOUT). The web interface now uses pretty printing for all PAGE related conversions by default.

**Comment by zuphilip** [comment]
Is this ready to merge? It looks good to me!

Do you agree that the problems with the alto files I described [above](https://github.com/UB-Mannheim/ocr-fileformat/pull/106#issuecomment-569919922) are upstream problems?

**Comment by stweil** [comment]
> There seems to be a side effect in the abbyy2hocr transformation. In the Web GUI this transformation output nothing with the input https://digi.bib.uni-mannheim.de/~stweil/ocr-praxis/Testseiten/abbyy/417576986_0031.xml .

Commit https://github.com/OCR-D/format-converters/commit/5b9568fd2b6dbfe891ef81826b7fffea7d21d814 was missing in our installation (fixed now). I noticed that just running `make all` or `make install` does not update any of the existing cloned code from external git repositories. I think we should start using git submodules to get explicit dependencies.

**Comment by stweil** [comment]
> I tried some examples in the Web GUI and see an error for alto__page transformation with https://rawgit.com/kba/ocr-fileformat-samples/master/samples/alto/2.0/wetzel_reisebegleiter_1901_0021.alto as well as with http://chroniclingamerica.loc.gov/lccn/sn86069133/1910-10-31/ed-1/seq-1/ocr.xml.

The first one throws a Java nullpointer exception, the second one looks like a vendor specific variant of ALTO. So both problems are not caused by the ocr-fileformat code.

**Comment by zuphilip** [comment]
> The first one throws a Java nullpointer exception, the second one looks like a vendor specific variant of ALTO. So both problems are not caused by the ocr-fileformat code.

Okay, we should report them upstream such that they will hopefully been fixed there in the future.

Moreover, I created an issue about a better update mechanism.

So, let me ask again: Is this ready to merge? It looks good to me! :+1:

**Comment by zuphilip** [comment]
Thank you @stweil for all the work on this! :bowing_man:

File: bin/ocr-transform.sh [hunk 1]
```diff
@@ -50,6 +49,8 @@ main () {
         fi
     fi
 
+    declare -a script_args
```
Comment (kba) [inline bin/ocr-transform.sh]: Why did you move that line?

File: bin/ocr-transform.sh [hunk 2]
```diff
@@ -86,8 +87,7 @@ main () {
         [[ "$outfile" != '-' ]] &&  script_args=("${script_args[@]}" "-o:$outfile")
         exec_saxon "${script_args[@]}"
     else
-        script_args=("${script_args[@]}" "$infile")
-        script_args=("${script_args[@]}" "$outfile")
+        script_args=("$infile" "$outfile" "${script_args[@]}")
```
Comment (kba) [inline bin/ocr-transform.sh]: This is reverse sorted then before.

File: bin/ocr-transform.sh (hunk 1 above)
Comment (stweil) [reply bin/ocr-transform.sh]: I simply wanted to have it close to the first use of `script_args`.

File: bin/ocr-transform.sh (hunk 2 above)
Comment (stweil) [reply bin/ocr-transform.sh]: It is now sorted as described in [script/transform/README.md](https://github.com/UB-Mannheim/ocr-fileformat/blob/master/script/transform/README.md).

Comment (kba) [reply bin/ocr-transform.sh]: It was wrong before, weird we never noticed this...
